### PR TITLE
Js playercardcounters

### PR DESCRIPTION
**Pull request by jacobstringer**

### CODE REVIEW

**Comment by jacobstringer** [comment]
Changes Cardhist serialise and the print_json function to send back card history information of every player.

I will let @Marthijn-B merge this when he's happy that the front end is updated so this doesn't break anything.

Also now includes card colour counters.

@mmdonaldson, could you create a migration file for this PR and commit it? I'm still not able to do it.

**Review by mmdonaldson — CHANGES_REQUESTED** [review]
I will add the db migration script tonight. Blocking this PR until then incase we accidentally merge

**Comment by Marthijn-B** [comment]
Did my PR fail because there were alot of changes to the file?
ie, this branch is behind the master branch atm. 
This works on my machine... hmmm

**Comment by jacobstringer** [comment]
@Marthijn-B, it looks like the test failed because there's now no ./CardHist in PlayerDisplay.js. It'll require a bit of an update on that test I would say.

**Comment by Marthijn-B** [comment]
Ok, I see how to find that info now. Thanks.
You reckon I could just upload my CardHist.js file here, would fix that? (even though it has changed a little from the master branch now?

**Comment by mmdonaldson** [comment]
@Marthijn-B if I was you i'd merge master into your branch or rebase

**Comment by Marthijn-B** [comment]
What I used, was the current master (as of this afternoon)., added Jacobs changed files from this PR, and worked off that. 
Currently this PR status: "This branch is 5 commits ahead, 11 commits behind master."
CardHist.js in already in the master. (entered GitHub after this PR was created)
#Slightly confused

**Comment by mmdonaldson** [comment]
@Marthijn-B yeap it gets a bit chaotic when multiple people work on a branch. Once person will have to clean it all up

**Comment by mmdonaldson** [comment]
@Marthijn-B @jacobstringer happy to do it but just let me because only person can work on the branch to clean it up for merge

**Comment by jacobstringer** [comment]
Feel free to sort as needs be. I won't be doing any more back end work on this branch... it's mainly a case of getting the front end to be happy with the changes.



Sent from my Samsung Galaxy smartphone.

-------- Original message --------

**Comment by Marthijn-B** [comment]
All good. I wont do anymore to this branch

**Comment by Marthijn-B** [comment]
So I added CardHist.js into this PR.
Created conflicts in both CardHist and PlayerDisplay
Resolved all conflicts for these two files to the upto date code.
Lets see if circleci passes...

File: static/src/components/CardHist.js
Comment (mmdonaldson) [inline static/src/components/CardHist.js]: Can we remove console.log

File: static/src/components/CardHist.js
Comment (mmdonaldson) [inline static/src/components/CardHist.js]: Does the index of history start at 1?

Comment (Marthijn-B) [reply static/src/components/CardHist.js]: Starts at 1 because index zero is the wonder board itself. No need to display that, just the cards

File: static/src/components/CardHist.js
Comment (Marthijn-B) [reply static/src/components/CardHist.js]: Always one still slips thru. Removed the rest.

**Review by Marthijn-B — COMMENTED** [review]
CardHist.js changed

**Review by Marthijn-B — COMMENTED** [review]
Loop to start at index zero

**Review by mmdonaldson — APPROVED** [review]

**Pull request merged**
